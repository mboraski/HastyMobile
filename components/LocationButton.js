import React, { Component } from 'react';
import { StyleSheet, Platform } from 'react-native';
import { connect } from 'react-redux';
import { Icon } from 'react-native-elements';

import Color from '../constants/Color';
import { emY } from '../utils/em';

const SIZE = emY(1.5625);

class LocationButton extends Component {
    onPress = () => {};

    render() {
        return (
            <Icon
                name="navigation"
                size={SIZE}
                containerStyle={styles.iconContainer}
                iconStyle={styles.icon}
                onPress={this.onPress}
                raised
            />
        );
    }
}

const styles = StyleSheet.create({
    iconContainer: {
        backgroundColor: Color.GREY_100,
        marginRight: 12,
        ...Platform.select({
            ios: {
                shadowColor: '#000',
                shadowOffset: { width: 0, height: 2 },
                shadowOpacity: 0.3,
                shadowRadius: 1
            },
            android: {
                elevation: 4
            }
        })
    },
    icon: {
        color: Color.GREY_600,
        transform: [
            {
                rotate: '45deg'
            }
        ]
    }
});

const mapDispatchToProps = function (dispatch) {
    return {};
};

export default connect(null, mapDispatchToProps)(LocationButton);
