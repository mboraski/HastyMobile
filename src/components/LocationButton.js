import React, { Component } from 'react';
import { StyleSheet, Platform } from 'react-native';
import { Icon } from 'react-native-elements';

import Color from '../constants/Color';
import { emY } from '../utils/em';

const SIZE = emY(1.5625);

class LocationButton extends Component {
    render() {
        const { style, ...props } = this.props;
        return (
            <Icon
                {...props}
                name="navigation"
                size={SIZE}
                containerStyle={[styles.iconContainer, style]}
                iconStyle={styles.icon}
                raised
            />
        );
    }
}

const styles = StyleSheet.create({
    iconContainer: {
        backgroundColor: Color.GREY_100,
        marginRight: 12,
        ...Platform.select({
            ios: {
                shadowColor: '#000',
                shadowOffset: { width: 0, height: 2 },
                shadowOpacity: 0.3,
                shadowRadius: 1
            },
            android: {
                elevation: 4
            }
        })
    },
    icon: {
        color: Color.GREY_700,
        transform: [
            {
                rotate: '45deg'
            }
        ]
    }
});

export default LocationButton;
