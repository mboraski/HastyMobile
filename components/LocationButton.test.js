import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import LocationButton from './LocationButton';

const initialState = {};

describe('LocationButton', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<LocationButton />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
