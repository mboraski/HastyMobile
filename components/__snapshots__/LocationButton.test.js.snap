// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationButton renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Icon
    color="black"
    containerStyle={
        Object {
            "backgroundColor": "#F4F4F4",
            "marginRight": 12,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          }
    }
    iconStyle={
        Object {
            "color": "#4c4c4c",
            "transform": Array [
              Object {
                "rotate": "45deg",
              },
            ],
          }
    }
    name="navigation"
    onPress={[Function]}
    raised={true}
    reverse={false}
    reverseColor="white"
    size={25}
    underlayColor="white"
/>,
  "nodes": Array [
    <Icon
      color="black"
      containerStyle={
            Object {
                  "backgroundColor": "#F4F4F4",
                  "marginRight": 12,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
      }
      iconStyle={
            Object {
                  "color": "#4c4c4c",
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                }
      }
      name="navigation"
      onPress={[Function]}
      raised={true}
      reverse={false}
      reverseColor="white"
      size={25}
      underlayColor="white"
/>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <LocationButton />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LocationButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onPress": [Function],
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Icon
          color="black"
          containerStyle={
                    Object {
                              "backgroundColor": "#F4F4F4",
                              "marginRight": 12,
                              "shadowColor": "#000",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0.3,
                              "shadowRadius": 1,
                            }
          }
          iconStyle={
                    Object {
                              "color": "#4c4c4c",
                              "transform": Array [
                                Object {
                                  "rotate": "45deg",
                                },
                              ],
                            }
          }
          name="navigation"
          onPress={[Function]}
          raised={true}
          reverse={false}
          reverseColor="white"
          size={25}
          underlayColor="white"
/>,
        "_debugID": 4,
        "_renderedOutput": <Icon
          color="black"
          containerStyle={
                    Object {
                              "backgroundColor": "#F4F4F4",
                              "marginRight": 12,
                              "shadowColor": "#000",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0.3,
                              "shadowRadius": 1,
                            }
          }
          iconStyle={
                    Object {
                              "color": "#4c4c4c",
                              "transform": Array [
                                Object {
                                  "rotate": "45deg",
                                },
                              ],
                            }
          }
          name="navigation"
          onPress={[Function]}
          raised={true}
          reverse={false}
          reverseColor="white"
          size={25}
          underlayColor="white"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LocationButton />,
}
`;
